File: primo.py
from math import sqrt

primos = list()

n = 2

while True:
    add = True

    for p in primos:
        if p > sqrt(n):
            break

        if n % p == 0:
            add = False
            break

    if add:
        primos.append(n)
        print(n)

    n += 1

File: primo.go
package main

import (
	"fmt"
	"math"
)

func main() {
	primos := []int{}

	for n := 2; true; n++ {
		add := true

		for i := 0; i < len(primos); i++ {
			if primos[i] > int(math.Sqrt(float64(n))) {
				break
			}

			if n%primos[i] == 0 {
				add = false
				break
			}
		}

		if add {
			primos = append(primos, n)
			fmt.Println(n)
		}
	}
}
